src/sys/reboot.rs

```diff
@@ -1,48 +1,163 @@
+                /// command), so the system will reboot if a halt is not explicitly
+                /// requested.
+                #[cfg(target_os = "openbsd")]
+                RB_USERREQ;
+                /// Load the symbol table and enable a built-in debugger in the
+                /// system. This option will have no useful function if the kernel
+                /// is not configured for debugging. Several other options have
+                /// different meaning if combined with this option, although their
+                /// use may not be possible via the `reboot()` call. See `ddb(4)` for
+                /// more information.
+                RB_KDB;
+                /// Normally, the disks are sync'd (see `sync(8)`) before the
+                /// processor is halted or rebooted. This option may be useful
+                /// if file system changes have been made manually or if the
+                /// processor is on fire.
+                RB_NOSYNC;
+                /// Normally, the reboot procedure involves an automatic disk
+                /// consistency check and then multi-user operations. `RB_SINGLE`
+                /// prevents this, booting the system with a single-user shell on
+                /// the console. `RB_SINGLE` is actually interpreted by the `init(8)`
+                /// program in the newly booted system.
+                ///
+                /// When no options are given (i.e., `RB_AUTOBOOT` is used), the
+                /// system is rebooted from file /bsd in the root file system of
+                /// unit 0 of a disk chosen in a processor specific way. An automatic
+                /// consistency check of the disks is normally performed (see `fsck(8)`).
+                RB_SINGLE;
+                /// The `bootstr` argument is a string passed to the firmware
+                /// on the machine, if possible, if this option is set.
+                ///
+                /// Currently this is only implemented on the sparc and the sun3 ports.
+                #[cfg(target_os = "netbsd")]
+                RB_STRING;
+                /// Initially invoke the `userconf(4)` facility when the system
+                /// starts up again, if it has been compiled into the kernel
+                /// that is loaded.
+                #[cfg(target_os = "netbsd")]
+                RB_USERCONF;
+                /// Don't update the hardware clock from the system clock, presumably
+                /// because the system clock is suspect.
+                #[cfg(target_os = "openbsd")]
+                RB_TIMEBAD;
+            }
+        }
+
+        /// Reboot system or halt processor
+        ///
+        /// For more information, see [`reboot(2)`](https://man.netbsd.org/reboot.2).
+        #[cfg(target_os = "netbsd")]
+        pub fn reboot<S: AsRef<OsStr>>(how: RebootMode, bootstr: S) -> Result<Infallible> {
+            let bootstr = bootstr.as_ref().as_bytes().as_ptr().cast_mut().cast();
```

Comment:
If the underlying shutdown procedure will modify `bootstr`, then we will get a UB here. I am not sure if `reboot(2)` would change it or not, the manual does not mention this:

> `bootstr` is a string passed to the firmware on the machine, if possible, if this option is set.  Currently this is only implemented on the sparc and the sun3 ports.

But practically, it has little point in writing to that piece of memory given that everything will be gone after shutdown.

Reply:
The man page seems to suggest that bootstr is entirely ignored unless `RB_STRING` is specified, and that can only happen on sparc64.  So I suggest that we simply remove this argument and the `RB_STRING` flag until such a time as there is a least one known user of Nix on sparc64-unknown-netbsd.

Reply:
argument removed